mapiproxy/servers/default/emsmdb/oxcmsg.c

```diff
@@ -1485,7 +1486,9 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopCreateAttach(TALLOC_CTX *mem_ctx,
 								   messageID, message_object);
 		if (attachment_object) {
 			mretval = mapistore_message_create_attachment(emsmdbp_ctx->mstore_ctx, contextID, message_object->backend_object,
-								      attachment_object, &attachment_object->backend_object, &mapi_repl->u.mapi_CreateAttach.AttachmentID);
+								      attachment_object, &attachment_object->backend_object, &attachmentID);
+			attachment_object->object.attachment->attachmentID = attachmentID;
```

Comment:
it would make more sense to assign `attachmentID` after checking `mretval` is ok